### DISCUSSION TITLE
Is it possible to override Less variables (eg primary/seconadry color) in Blazor app?

### DISCUSSION THREAD

**Discussion by TarasOnyshko** [ant-design-blazor/ant-design-blazor#2380] [Q&A]
Hello everyone. I'm looking for a solution to override default styles in my Blazor app. Ideally, I would like just to override some Less variables to change the colors. 
I'm using the next styles "_content/AntDesign/css/ant-design-blazor.css"

Currently, there is a [documentation](https://ant.design/docs/react/customize-theme)  on how to do that for React apps, but I can not find any suitable way for the Blazor case. 
Also would be great to have the possibility to update styles on the code level as [Blazorise](https://blazorise.com/docs/theming) offers, but I'm not sure if that is available in AntDesign. 

Any ideas are appreciated, thanks!

**Comment by ElderJames** [comment]
I prefer CSS variables. Please see https://ant.design/docs/react/customize-theme-variable. 

We just need to solve the problem of less compilation, which we haven't solved yet.

**Reply by TarasOnyshko** [reply]
@ElderJames Thanks for your reply. Is there some code example/repo on how to do a trick to override the primary color of default AntDesign styles in Blazor server-side app? I saw that I may need to re-compile the Less file, but I'm not sure how to do that because of my lack of experience. 
Or please let me know if there is another way to change default styling, thank you.

**Reply by lukblazewicz** [reply]
@TarasOnyshko 
You can override the defaults if you use `antd` styles via webpack according to the following documentation:
https://ant.design/docs/react/customize-theme#Customize-in-webpack
It is possible as `AntBlazor` styles are kept in sync with the react counterpart, so that raw `antd` styles compiled customized values work perfectly well (and can replace the styles provided by `AntBlazor`). 
You can find slightly more info here:
https://github.com/ant-design-blazor/ant-design-blazor/issues/2174

**Comment by ElderJames** [comment]
We have successfully fixed less compilation to support CSS variables in the next release. 
You can then implement code-level changes to some variables.
